### DISCUSSION TITLE
Smart Cross-ref and Linking Issues

### DISCUSSION THREAD

**Discussion by andriandreo** [ahrm/sioyek#826] [Q&A] [answered]
For cites using the " [1], [1,2] " style, smart linking works like a charm (except for " [1-5] " in most cases...). However, I have faced this issue for several articles so far: whenever the citing style is using superscripts such as " <sup>1</sup>, <sup>1-5</sup>, <sup>1,2</sup> " with no explicit on-document linking, Sioyek is unable to resolve them. The same occurs if 'Fig. 1 and 2' is used instead of 'Figure 1' and 'Figure 2'. 

Moreover, even when explicit on-document linking exists, the righ-click preview shows a portion of document which does not correspond to the cross-ref. Middle-click starts a Google Scholar search of the text  right after the " [1-5] " or " <sup>1-5</sup> " cite, which is pretty useless.

I am aware this is not an easy project and I am really grateful and impressed with the work so far, just wanted to help offering a bit of debugging.

The issue occurs with `sioyek 2.0.0` release in Arch Linux for both `sioyek AUR pkg` and GitHub `sioyek AppImage` release.

**Comment by ahrm** [comment]
> Moreover, even when explicit on-document linking exists, the righ-click preview shows a portion of document which does not correspond to the cross-ref.

This should not be the case.  This probably means that the link in the PDF file is wrong (that is, the link itself is pointing to the wrong location). If this is not the case, then this is a bug. Can you share a PDF with such issue?

**Reply by andriandreo** [reply]
> This probably means that the link in the PDF file is wrong (that is, the link itself is pointing to the wrong location)

I'm afraid not... Please, find below the two articles I'm facing the mentioned issues at the moment (they should be Open Access, otherwise let me know and I'll send via email):

- https://pubs.rsc.org/en/content/articlelanding/2021/CP/D1CP00661D ('Fig. 1 and 2' and '<sup>1-5</sup>' ref/citing)
- https://pubs.acs.org/doi/full/10.1021/acsaelm.1c00116 (Middle-click over cites leads to Google Scholar search; right-click points to wrong preview)

Thanks!

**Reply by ahrm** [reply, marked as answer]
You are right. However, right clicking on links still should work on most documents. The reason that it doesn't work on this document is that it uses a relatively obscure link format (viewrect instead of xyz) which we don't support (I didn't know it existed!). It will be supported in the next version of sioyek. Thanks for bringing that to my attention.

**Reply by ahrm** [reply]
However, note that the links in the document are still sometimes wrong (in addition to being in an unusual format)! For example many of the links to references point to the exact same location.

**Reply by andriandreo** [reply]
Thanks to you! 
And you are right, some links are not working properly in the second paper, especially regarding Figures. 

The right-click (smart) preview for figures in the first paper is not working properly either (where no on-document explicit links are provided).
